src/modules/paymentProcessor/PP_Simple_v1.sol

```diff
@@ -142,4 +159,54 @@ contract PP_Simple_v1 is Module_v1, IPaymentProcessor_v1 {
         // Since we pay out on processing, this function does nothing
         return;
     }
+
+    /// @inheritdoc IPaymentProcessor_v1
+    function unclaimable(address client, address token, address paymentReceiver)
+        public
+        view
+        returns (uint amount)
+    {
+        return unclaimableAmountsForRecipient[client][token][paymentReceiver];
+    }
+
+    /// @inheritdoc IPaymentProcessor_v1
+    function claimPreviouslyUnclaimable(
+        address client,
+        address token,
+        address receiver
+    ) external {
+        if (unclaimable(client, token, _msgSender()) == 0) {
+            revert Module__PaymentProcessor__NothingToClaim(
+                client, _msgSender()
+            );
+        }
+
+        _claimPreviouslyUnclaimable(client, token, receiver);
+    }
+
+    /// @notice used to claim the unclaimable amount of a particular paymentReceiver for a given payment client
+    /// @param client address of the payment client
+    /// @param token address of the payment token
+    /// @param paymentReceiver address of the paymentReceiver for which the unclaimable amount will be claimed
+    function _claimPreviouslyUnclaimable(
+        address client,
+        address token,
+        address paymentReceiver
+    ) internal {
+        //get amount
+
+        address sender = _msgSender();
+        //copy value over
+        uint amount = unclaimableAmountsForRecipient[client][token][sender];
+        //Delete the field
+        delete unclaimableAmountsForRecipient[client][token][sender];
+
+        //Call has to succeed otherwise no state change
+        IERC20(token).safeTransferFrom(client, paymentReceiver, amount);
+
+        emit TokensReleased(paymentReceiver, address(token), amount);
+
+        //Make sure to let paymentClient know that amount doesnt have to be stored anymore
+        IERC20PaymentClientBase_v1(client).amountPaid(token, amount);
```

Comment:
Would moving this up to before the transfer alter functionality? To avoid any weird callback shenanigans with malicious tokens that then call the client.